### ISSUE TITLE
shell_plus interactive shell doesn't work properly in windows10?

### ISSUE THREAD

**Issue by rightx2** [django-extensions/django-extensions#1470]
`shell_plus` doesn't work well in windows 10 (using `git-bash` application`)
- `tab` doesn't support completion (It just works as `tab`, which is 4 spaces indentation, even after `my_variable.<tab>`)
- Interactive shell line is not colorful (just grey)
- input cursor can go anywhere......

Belows are my trials:

1. `django-extensions == 2.1.1` & `python manage.py shell_plus`
2. `django-extensions == 2.26.` & `python manage.py shell_plus --ipython`

Phenomenons:

![Screen Shot 2020-01-26 at 13 02 03](https://user-images.githubusercontent.com/10606994/73130496-a80f1e80-403c-11ea-8467-58b8703ae853.png)

Interesting thing is that 
1. With same trials in Linux(Ubuntu 16.04) it works well..
2.` python manage.py shell_plus --notebook` works well in jupyter
2. When running `ipython` in the shell(not `shell_plus`), then interactive shell works well as it meant to be(supporting tab autocompletion etc)

Why does it happen?

**Comment by rightx2** [comment]
![Screen Shot 2020-01-26 at 13 11 19](https://user-images.githubusercontent.com/10606994/73130548-60d55d80-403d-11ea-9d22-45cddb34f47b.png)

**Comment by trbs** [comment]
I am sorry but I can't help you very much :( 
I do not own nor use any Windows machines...

Reasoning about it I would say that `shell_plus` does not do anything special with the underlying REPL. Although there clearly is some difference if your `ipython` / `bpython` / `pypython` / etc shells are functioning correct otherwise.

Do you have similar kind of issues with vanilla `./manage.py shell_plus` with `ipython` installed ?  (eg: the disappearing color highlighting)

If so it might be something to do with `django` commands on windows 10 ?

**Comment by rightx2** [comment]
Same issues happened with vanilla `./manage.py shell_plus` : (

I don't think this is related with django command because other commands work pretty well

**Comment by trbs** [comment]
Sorry I meant `./manage.py shell` (the standard Django one) specifically comparing `ipython` in both `shell` and `shell_plus` to see it it's `django-extensions` specific.

**Comment by rightx2** [comment]
Oops,
1. `manage.py shell`: issues
2. `manage.py shell_plus`: issues 
3. `ipython`: works well

So, it indicates that this is not just `django-extensions` issues, right?

**Comment by trbs** [comment]
Yeah think this might be related to something in `Django` management commands itself.
If it fails in `manage.py shell` I would not expect `shell_plus` to do any better... so maybe also open an issue with `Django` itself as well.

That said if there is a fix we can do here I'm happy to merge a PR.

**Comment by Ivin0022** [comment]
I had the similar issue with git-bash on windows 10, fixed it by using git-bash inside vscode

**Comment by cyber778** [comment]
Happened to me on python 3.7, when I move back to 3.5 it works again...

**Comment by trbs** [comment]
That's a shame, since you should use Python 3.7 (or even better 3.8) and not the much older 3.5 :-)

Sorry I do not own any Windows to test this out... a PR with test case would be much appreciated.

(Hint: if you are a company that have employee's who are bitten by this please sponsor our work here so we can try and fix this.)
